packages/components/src/navigation/stories/index.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import { Navigation } from '..';
+import { NavigationBackButton } from '../back-button';
+import { NavigationGroup } from '../group';
+import { NavigationItem } from '../item';
+import { NavigationMenu } from '../menu';
+import { DefaultStory } from './utils/default';
+import { GroupStory } from './utils/group';
+import { ControlledStateStory } from './utils/controlled-state';
+import { SearchStory } from './utils/search';
+import { MoreExamplesStory } from './utils/more-examples';
+import { HideIfEmptyStory } from './utils/hide-if-empty';
+import './style.css';
+
+const meta: ComponentMeta< typeof Navigation > = {
```

Comment:
I tried to apply the least amount of effort required to get controls and auto-generated docs working correctly.

The stories themselves would also benefit from a refactor, but it's out of scope for this PR (plus, we are encouraging folks to use `Navigator` instead of this component)